build.gradle

```diff
@@ -196,9 +203,12 @@ def Task getPublishArtifactsTask(String buildVersion, Project subproject) {
 allprojects {
     apply plugin: 'com.bmuschko.docker-remote-api'
 
-    task copyDocker(type: Sync) {
-        from "${project.projectDir}/Dockerfile"
```

Comment:
So if the behavior of the tasks different from OSS to cloud, wouldn't one override the other in this approach? I worry that if that were to happen (even though these two tasks are currently identical) you might end up in a situation where the wrong task was used unexpectedly.

Should we instead ensure that there are no duplicate names (maybe by just letting the build fail) and rename the OSS or cloud tasks and update the builds where they get called??

I know at least listAllDependencies isn't called by any builds (I wrote that one ;) )

Reply:
> So if the behavior of the tasks different from OSS to cloud, wouldn't one override the other in this approach?

When building `oss`, the `oss` one will be used. When building `cloud`, the `cloud` one will be used (as `cloud` will load first and only `oss` has this check).  This is the behavior we want.

Maybe they should have different names, but that is something we can address after the merge of `oss` into `cloud`. I'm trying to keep breaking changes to a minimum prior to the merge.

Reply:
can you add a comment explaining this nuance? will be easier to misunderstand otherwise.

Reply:
@davinchia I'm planning on doing this better (something along the lines of what Conor recommended) after the code has been merged.  So this nuance should only be around for about a week (famous last words, I know).